### ISSUE TITLE
rsyncing the BEAKERLIB_DIR back and forth may break tmt run --last

### ISSUE THREAD

**Issue by sopos** [teemtee/tmt#502]
BEAKERLIB_DIR is kind of beakerlib's internal data store which can contain virtually anything in future.
Due to a small quirk in beakerlib the root backup dir gets permissions of `/` wich does not have write permisson, see output of `ls -dl /`.
This means that if beakerlib does a `rlFileBackup` the next `tmt run --last` fails on permission denied when cleaning the workdir.
```
$ tmt run --last test --name . discover -f execute -f --how tmt --interactive
/var/tmp/tmt/run-059

/plans/default
Traceback (most recent call last):
  File "/usr/bin/tmt", line 8, in <module>
    tmt.cli.main()
  File "/usr/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.9/site-packages/click/core.py", line 1290, in invoke
    return _process_result(rv)
  File "/usr/lib/python3.9/site-packages/click/core.py", line 1224, in _process_result
    value = ctx.invoke(self.result_callback, value, **ctx.params)
  File "/usr/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3.9/site-packages/tmt/cli.py", line 280, in finito
    click_context.obj.run.go()
  File "/usr/lib/python3.9/site-packages/tmt/base.py", line 990, in go
    plan.go()
  File "/usr/lib/python3.9/site-packages/tmt/base.py", line 565, in go
    step.wake()
  File "/usr/lib/python3.9/site-packages/tmt/steps/execute/__init__.py", line 87, in wake
    super().wake()
  File "/usr/lib/python3.9/site-packages/tmt/steps/__init__.py", line 121, in wake
    self._workdir_cleanup()
  File "/usr/lib/python3.9/site-packages/tmt/utils.py", line 475, in _workdir_cleanup
    shutil.rmtree(directory)
  File "/usr/lib64/python3.9/shutil.py", line 718, in rmtree
    _rmtree_safe_fd(fd, path, onerror)
  File "/usr/lib64/python3.9/shutil.py", line 655, in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
  File "/usr/lib64/python3.9/shutil.py", line 655, in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
  File "/usr/lib64/python3.9/shutil.py", line 655, in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
  [Previous line repeated 3 more times]
  File "/usr/lib64/python3.9/shutil.py", line 659, in _rmtree_safe_fd
    onerror(os.rmdir, fullname, sys.exc_info())
  File "/usr/lib64/python3.9/shutil.py", line 657, in _rmtree_safe_fd
    os.rmdir(entry.name, dir_fd=topfd)
PermissionError: [Errno 13] Permission denied: 'etc'
```

This may be fixed by simple patch on beakerlib side.
```
$ git diff
diff --git a/src/infrastructure.sh b/src/infrastructure.sh
index d0493de..21c7733 100644
--- a/src/infrastructure.sh
+++ b/src/infrastructure.sh
@@ -747,12 +747,12 @@ rlFileBackup() {
         dir="$path"
         failed=false
         while true; do
+            [ "$dir" == "/" ] && break
             $acl && { getfacl --absolute-names "$dir" | setfacl --set-file=- "${backup}${dir}" || failed=true; }
             $selinux && { chcon --reference "$dir" "${backup}${dir}" || failed=true; }
             chown --reference "$dir" "${backup}${dir}" || failed=true
             chmod --reference "$dir" "${backup}${dir}" || failed=true
             touch --reference "$dir" "${backup}${dir}" || failed=true
-            [ "$dir" == "/" ] && break
             dir=$(dirname "$dir")
         done
         if $failed; then
```
But still, the question is why to copy back and forth so much data if basically just a `journal.xml` and a `TestResults` files are those interested on the way back from SUT? There might be even few gigs of data, depending on how wrongly the test is written.

**Comment by sopos** [comment]
Actually the patch does not prevent this completely as there might be some path component intentionally created without write permissions, but it will probably fix the issue in most of the cases. Anyway I would like to see some updated on tmt side.

**Comment by lukaszachy** [comment]
> But still, the question is why to copy back and forth so much data if basically just a `journal.xml` and a `TestResults` files are those interested on the way back from SUT? There might be even few gigs of data, depending on how wrongly the test is written.

IMHO this could be covered by #280 - E.g. framework:beakerlib should add two artifacts: journal and testresults so tmt will know what to copy back.

**Comment by sopos** [comment]
> So this is the reason why #283 happens?

exactly

**Comment by psss** [comment]
This seems to be quite an important issue. Proposing for 1.5.

**Comment by sopos** [comment]
Also `TestResults`. The `journal.txt` may be useful but not required.

**Comment by lukaszachy** [comment]
Hopefully Autumn will bring more time to work on this. Moving to 1.9

**Comment by psss** [comment]
Notes from the hacking session: Possible approach could be to use `rsync` options not to try to preserve ownership. This could be a simple/fast change. And as the next step we could improve even more to skip unnecessary files completely.

**Comment by psss** [comment]
As the first step, what about excluding the `backup` directory during `guest.pull()` after test is executed? This should be fairly simple to implement and could fix problem with permission issues caused by backed-up system files.

**Comment by sopos** [comment]
for the beakerlib framework the files to be synced back from individual BEAKERLIB_DIRs are:
_TestResults_,
_journal.xml_,
and possibly _journal.txt_
\+ files generated by `rlFileSubmit` and `rlBundleLogs`

**Comment by lukaszachy** [comment]
If we are willing to accept just excluding 'backup' directory as fix then #1239 will fix this.
However there is not explicit enumeration of files necessary for beakerlib  as well as rlFilesubmit + TMT_TEST_DATA.

**Comment by lukaszachy** [comment]
Please reopen if there is more work necessary than #1239  and #1254 (both merged)
